src/providers/UserProvider.tsx

```diff
@@ -8,5 +13,30 @@ export type UserProviderProps = {
 };
 
 export default function UserProvider({ children, data }: UserProviderProps): JSX.Element {
+  const snackbar = useSnackbar();
+
+  useEffect(() => {
+    const populateTimeZone = async () => {
+      if (data.user) {
+        if (!data.user.timeZone) {
+          // Should validate bowser time zone with supported time-zones returned by backend
+          const browserTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+          const updateResponse = await updateUserProfile({ ...data.user, timeZone: browserTimeZone });
+          if (updateResponse.requestSucceeded) {
+            data.reloadUser();
+            snackbar.pageSuccess(
+              strings.formatString(
+                strings.UPDATED_TIMEZONE_MSG,
+                <Link to={APP_PATHS.MY_ACCOUNT}>{strings.MY_ACCOUNT}</Link>
+              ),
+              strings.formatString(strings.UPDATED_TIMEZONE_TITLE, browserTimeZone)
+            );
+          }
+        }
+      }
+    };
+    populateTimeZone();
```

Comment:
can we call this only if the time zone opt-in feature is enabled?

Reply:
Right!